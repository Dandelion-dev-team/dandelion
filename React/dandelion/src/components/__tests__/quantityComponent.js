import React from "react"
import renderer from "react-test-renderer"
import QuantityComponent from "../quantityComponent"


describe("Quantity", () => {
    let originalFetch;

    beforeEach(() => {
        originalFetch = global.fetch;
        global.fetch = jest.fn(() => Promise.resolve({
            json: () => Promise.resolve({
                "quantities": [
                    {
                      "id": 0,
                      "name": "Temperature",
                      "unit": "ÂºC",
                      "help_url": "https://en.wikipedia.org/wiki/Celsius"
                    },
                    {
                      "id": 1,
                      "name": "Humidity",
                      "unit": "%",
                      "help_url": "https://en.wikipedia.org/wiki/Humidity"
                    }
                  ],
            })
        }));
    });

    afterEach(() => {
        global.fetch = originalFetch;
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(<QuantityComponent/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})

