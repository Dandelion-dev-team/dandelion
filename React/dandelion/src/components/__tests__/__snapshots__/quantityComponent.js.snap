// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quantity renders correctly 1`] = `
<div
  className="authTable"
>
  <table
    className="tableList"
  />
</div>
`;
